tests/unit/amo/components/TestReportAbuseButton.js

```diff
@@ -0,0 +1,275 @@
+  });
+
+  it('shows more content when the "report" button is clicked', () => {
+    const fakeEvent = createFakeEvent();
+    const root = renderMount();
+
+    root.find('.ReportAbuseButton-show-more').simulate('click', fakeEvent);
+
+    sinon.assert.called(fakeEvent.preventDefault);
+    expect(root.find('.ReportAbuseButton--is-expanded')).toHaveLength(1);
+  });
+
+  it('dismisses more content when the "dismiss" button is clicked', () => {
+    const fakeEvent = createFakeEvent();
+    const root = renderMount();
+
+    root.find('.ReportAbuseButton-show-more').simulate('click');
+
+    root.find('.ReportAbuseButton-cancel-report').simulate('click', fakeEvent);
+
+    sinon.assert.called(fakeEvent.preventDefault);
+    expect(root.find('.ReportAbuseButton--is-expanded')).toHaveLength(0);
+  });
+
+  it('disables the submit button if there is no abuse report', () => {
+    const root = renderMount();
+
+    expect(root.find('.ReportAbuseButton-send-report')).toHaveProp('disabled');
+  });
+
+  it('disables the submit button if text in the textarea is removed', () => {
+    const root = renderMount();
+
+    // This simulates entering text into the textarea.
+    const textarea = root.find('.ReportAbuseButton-textarea textarea');
+    textarea.node.value = 'add-on ate my homework!';
+    textarea.simulate('change');
+
+    expect(root.find('.ReportAbuseButton-send-report').prop('disabled'))
+      .toEqual(false);
+
+    textarea.node.value = '';
+    textarea.simulate('change');
+
+    expect(root.find('.ReportAbuseButton-send-report').prop('disabled'))
+      .toEqual(true);
+  });
+
+  it('disables the buttons while sending the abuse report', () => {
+    const addon = { ...fakeAddon, slug: 'bank-machine-skimmer' };
+    const fakeEvent = createFakeEvent();
+    const { store } = dispatchClientMetadata();
+    store.dispatch(sendAddonAbuseReport({
+      addonSlug: addon.slug,
+      errorHandlerId: 'my-error',
+      message: 'All my money is gone',
+    }));
+    const root = renderMount({ addon, store });
+
+    // Expand the view so we can test that it wasn't contracted when
+    // clicking on the disabled "dismiss" link.
+    root.find('.ReportAbuseButton-show-more').simulate('click');
+    expect(root.find('.ReportAbuseButton--is-expanded')).toHaveLength(1);
+
+    const cancelButton = root.find('.ReportAbuseButton-cancel-report');
+    const sendButton = root.find('.ReportAbuseButton-send-report');
+
+    expect(cancelButton)
+      .toHaveClassName('ReportAbuseButton-cancel-report--disabled');
+    expect(sendButton.prop('disabled')).toEqual(true);
+    expect(sendButton.prop('children')).toEqual('Sending abuse report');
+
+    cancelButton.simulate('click', fakeEvent);
+    sinon.assert.called(fakeEvent.preventDefault);
+    expect(root.find('.ReportAbuseButton--is-expanded')).toHaveLength(1);
+  });
+
+  it('shows a success message and hides the button if report was sent', () => {
```

Comment:
English is annoying, but generally in conversation one would say "the letter was sent" not "the letter is sent". Similarly, one would say "The letter was not sent" instead of "the letter is not sent". Even with "yet", it'd be "the letter has not yet been sent" instead of "the letter is not sent yet".

Really this would be nicer if I wrote "if the report was sent" but I wanted to fit it on 80 chars. 😆

Reply:
Ah ok, sorry for the false positive then. I was thinking about sequence of tenses..

Reply:
No worries, it's confusing! 🤣